content/3_Silver/Binary_Search.mdx

```diff
@@ -112,7 +112,8 @@ range `[lo,hi]`.
 using namespace std;
 
 int lastTrue(int lo, int hi, function<bool(int)> f) {
-	for (--lo; lo < hi; ) {
```

Comment:
wait why is this hard to read? what's wrong with for loops D:

Reply:
the intent behind it is kinda unclear

Reply:
isn't this what the initialization block of a for loop is for?

Reply:
yeah but for loops aren't usually used in this manner (i haven't seen them even in many cp solutions)

Reply:
I think this was fine before too... (I use for loops like that all the time, and it's very much intended behavior)

Reply:
don't think the average person will be accustomed to for loops being used like this though

Reply:
well, good time to learn then :)

Reply:
then like put a comment indicating what it does

Reply:
I wouldn't expect the average student to understand that for loop syntax.

Reply:
Isn't this basic C++ for loop stuff?

Reply:
> well, good time to learn then :)

^^

Reply:
no it isn't really basic